### ISSUE TITLE
Exposing methods for Script Setup 'SFC Composition API Syntax Sugar'

### ISSUE THREAD

**Issue by ux-engineer** [intlify/vue-i18n#138]
Vue 3 has this yet experimental syntax [Script Setup 'SFC Composition API Syntax Sugar'](https://github.com/vuejs/rfcs/blob/sfc-improvements/active-rfcs/0000-sfc-script-setup.md).

I'm wondering how to best expose the translation method `t` in components written with this syntax.

Currently this syntax is working for me:

```vue
<script setup lang="ts">
import { useI18n } from 'vue-i18n';

const { t } = useI18n();

export const name = 'ViewName';

// Note: vue-meta package does not support Vue 3 as of yet, but am verifying functionality
// by being able to use this object in template syntax.
export const metaInfo = {
  title: t('PAGE_TITLE'),
  meta: [
    { name: 'description', content: t('PAGE_DESCRIPTION') },
  ],
};
</script>

<template>
  <h1>{{ metaInfo.title }}</h1>
</template>
```

However, **I would like to be able to write it like this**:

```vue
<script setup lang="ts">
export const name = 'ViewName';

export const metaInfo = {
  title: $t('PAGE_TITLE'),
  meta: [
    { name: 'description', content: $t('PAGE_DESCRIPTION') },
  ],
};
</script>
```

Would it be possible for this library to automatically expose $t method in script setup scope? Or by object-destructuring from setup context `<script setup="_, { $t }" lang="ts">`?

**Comment by natemate90** [comment]
The global exposure of the `t()` would be a great improvement, as I understand it, `$t()` cannot make use of the typed translation schema?

**Comment by Liwoj** [comment]
Note that `script setup` proposal linked in OP was dropped. Current Vue 3 `script setup` is based on the [new proposal](https://github.com/vuejs/rfcs/blob/master/active-rfcs/0040-script-setup.md) which has no `setup` context

If I understand it correctly, auto exposing composer functions would require tapping into SFC compiler or writing bundler plugin similar to [unplugin-auto-import](https://github.com/antfu/unplugin-auto-import) and I'm not convinced the benefits outweigh the complexity of the solution...especially when you can use above mentioned plugin to completely eliminate the `import` statement (plugin has already [vue-i18n](https://github.com/antfu/unplugin-auto-import/blob/main/src/presets/vue-i18n.ts) [preset](https://github.com/antfu/unplugin-auto-import#configuration))

**Comment by Liwoj** [comment]
@natemate90 Nope. `unplugin-auto-import` can only do the import part (`import { useI18n } from 'vue-i18n';`). You still need to **call it** - (it's not import) `const { t } = useI18n();` to get a `t` (or `d` etc)

**Comment by omnjpa27516** [comment]
Is there any update on this issue? @Liwoj

**Comment by Liwoj** [comment]
@omnjpa27516 Nope, my last comment is still valid and I don't think it will change (like ever :))

**Comment by kazupon** [comment]
Sorry late my reply

As Liwoj pointed out, the original `<script setup>` RFC was replaced, and auto-exposing functions in `<script setup>` would require tapping into the SFC compiler, which is outside vue-i18n's scope.

In v12, `$t` is available in **templates** by default (`globalInjection: true` is the default), so no `useI18n()` call is needed for template usage:

```vue
<template>
  <!-- Works without useI18n() -->
  <h1>{{ $t('PAGE_TITLE') }}</h1>
</template>
```

For `<script setup>`, `useI18n()` is still required — this is by design, consistent with how Vue composables work (e.g., `useRouter()`, `useStore()`):

```vue
<script setup>
import { useI18n } from 'vue-i18n'
const { t } = useI18n()
</script>
```

If you want to eliminate the `import` statement, you can use [unplugin-auto-import](https://github.com/antfu/unplugin-auto-import) with the vue-i18n preset to auto-import `useI18n`. The `const { t } = useI18n()` call is still needed, but this is a one-liner per component.

Closing this as this is a Vue platform constraint, not something vue-i18n can change.
Thanks!
